### PR TITLE
51 analysis module first time series analysis implemented

### PR DESCRIPTION
**Pull request by heskil**
first and tentative implementation of a time series analysis. requires further experimentation

### CODE REVIEW

File: apps/analyzer/metadata_analyzer/main.py [hunk 1]
```diff
@@ -346,12 +349,200 @@ def simple_rule_based_analysis_storage_capacity():
         return "Invalid value for alert limit", 400
 
 
+
+# TODO yaml for swagger
+@app.route("/kMeansAnomalies", methods=["POST"])
+def runTimeSeriesTests():
+    """Runs k-means anomaly detection on the specified dataset.
```
Comment (engelharddirk) [inline apps/analyzer/metadata_analyzer/main.py:356]: Just a side note: I get the feeling that the in-code swagger YAML ducumentation is getting slightly out of hand :D
As far as I know, in flasgger, you can [use external yaml files](https://github.com/flasgger/flasgger?tab=readme-ov-file#using-external-yaml-files) to keep the code more tidy. Maybe that is something we could try?

File: apps/analyzer/metadata_analyzer/main.py [hunk 2]
```diff
@@ -346,12 +349,200 @@ def simple_rule_based_analysis_storage_capacity():
...
+    except KeyError:
+        return "Missing field of type " + field
+
+    try:
+        result = Analyzer.simple_time_series_analysis(
+            variable, task_id, frequency, backup_type, window_size
+        )
+        return jsonify(result)
+    except ValueError as val:
+        return "Value error occured: " + str(val), 401
```
Comment (engelharddirk) [inline apps/analyzer/metadata_analyzer/main.py:436]: As we have discussed earlier, this error code is not representing what is happening

**Review by engelharddirk — APPROVED** [review]
As far as I can tell, it looks good! Only left very few side notes.
dev has still to be merged, but when the merge conflicts are solved and the technical issues are adressed, we're good to go, I think :)

File: apps/analyzer/metadata_analyzer/main.py (hunk 2 above)
Comment (heskil) [reply apps/analyzer/metadata_analyzer/main.py:436]: Done and changed :)

**Pull request merged**